Content.Server/Power/Pow3r/BatteryRampPegSolver.cs

```diff
@@ -252,8 +252,8 @@ private void UpdateNetwork(Network network, PowerState state, float frameTime)
                 // available supply. IMO this is undesirable, but I can't think of an easy fix ATM.
 
                 battery.CurrentStorage -= frameTime * battery.CurrentSupply;
-                DebugTools.Assert(battery.CurrentStorage >= 0 || MathHelper.CloseTo(battery.CurrentStorage, 0, 1e-5));
                 battery.CurrentStorage = MathF.Max(0, battery.CurrentStorage);
+                DebugTools.Assert(battery.CurrentStorage >= 0 || MathHelper.CloseTo(battery.CurrentStorage, 0, 1e-5));
```

Comment:
The point of the assert is to check that (up to some floating point rounding errors, the result is correct. So the check should be before the clamping, otherwise that defeats the whole point.

If the error tolerance is too low here, then the assert should just be replaced with something like

```cs
#if DEBUG
                if (battery.CurrentStorage < 0)
                {
                    // Manual "MathHelper.CloseToPercent" using the subtracted value as the relative error source
                    float epsilon = Math.Max(frameTime * battery.CurrentSupply, 1) * 1e-4f;
                    DebugTools.Assert(battery.CurrentStorage > - epsilon);
                }
#endif
```

If that still fails, it might be because of some actual error somewhere (and not simply floating point precision)